### ISSUE TITLE
Docs: v2 todo list

### ISSUE THREAD

**Issue by provinzkraut** [litestar-org/litestar#1421]
This is a tracking issue for tasks related to documentation updates that need to happen before the v2 release.

- [x] #1285
- [x] #1050 
- [ ] [the OpenAPI generation CLI doc](https://docs.starliteproject.dev/2/usage/cli.html#openapi) isn't up to date with the current CLI: the new command is `starlite schema typescript` (for instance, to get the typescript client lib)
- [ ] [the tutorial on JWT Auth uses the old typing API for the `Request` type](https://docs.starliteproject.dev/2/usage/security/abstract-authentication-middleware.html#example-implementing-a-jwtauthenticationmiddleware). The V2 type is `Request[User,Token,State]`.
- [x] #1429
- [x] #1619

<a href="https://polar.sh/litestar-org/litestar/issues/1421">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/1421/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/litestar-org/litestar/issues/1421/pledge.svg">
</picture>
</a>

**Comment by mikkelsvartveit** [comment]
[The docs for AbstractAuthenticationMiddleware](https://docs.litestar.dev/2/usage/security/abstract-authentication-middleware.html) has these imports at the very top:

```
from litestar.authentication import (
    AbstractAuthenticationMiddleware,
    AuthenticationResult,
)
```

It seems like the correct import for these is now:

```
from litestar.middleware import (
    AbstractAuthenticationMiddleware,
    AuthenticationResult,
)
```

**Comment by ctrl-Felix** [comment]
> [The docs for AbstractAuthenticationMiddleware](https://docs.litestar.dev/2/usage/security/abstract-authentication-middleware.html) has these imports at the very top:

Fixed in #1814
